__tests__/integration/app.test.js

```diff
@@ -0,0 +1,56 @@
+const app = require("../../app");
+const request = require("supertest");
+const fact = require("../../models/fact");
+const api = require("../../routes/api");
+
+describe("Integration/unit testing app.js", function () {
+  describe("GET /", function () {
+    test("GET request responds with a 200 status code and returns HTML", async function (done) {
+      const response = await request(app).get("/");
+      const { statusCode, text } = response;
+      expect(statusCode).toBe(200);
+      expect(text).toContain("<!DOCTYPE html>");
+      done();
+    });
+
+    test("GET request should call fact.getFact()", async function (done) {
+      const getFact = jest.spyOn(fact, "getFact");
+      const response = await request(app).get("/");
+      expect(getFact).toHaveBeenCalled();
+      done();
+    });
+  });
+
+  describe("POST => /submit", function () {
+    test("POST request should respond with a 200 status code", async function (done) {
+      const response = await request(app).post("/submit");
+      const { statusCode } = response;
+      expect(statusCode).toBe(200);
```

Comment:
can you also make an assertion on the body? is there anything else in the response here to help clarify what it looks like?

Reply:
This one's a bit tricky. The post request for this route takes a request body and is supposed to return the same request body back to the user, but it's not enabled for the application yet. I just wrote this test to expect a 200 response code but I can make the changes if needed!

Reply:
I've enabled json for the express app, and changed the test!

Reply:
maybe just get rid of this test then, i think the test below is sufficient.

Reply:
okay, I just got rid of this test. Should I disable json for the application as well? @mmmaaatttttt

Reply:
@stephenisau  - i guess if you don't need it anymore, might as well remove.

Reply:
got it!